### PR TITLE
Update Microsoft.Authorization api version for Get-RoleDefinition

### PR DESCRIPTION
**Pull request by mumoryan**
## Description

Updates the RoleDefinition cmdlet api version to using 2022-05-01-preview

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Resources/Authorization.Management.Sdk/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+###
+``` yaml
+commit: 526e6049f46d58a5077850731dce19ab9767988f
+input-file:
+  - https://github.com/Azure/azure-rest-api-specs/blob/$(commit)/specification/authorization/resource-manager/Microsoft.Authorization/preview/2022-05-01-preview/authorization-RoleDefinitionsCalls.json
+  - https://github.com/Azure/azure-rest-api-specs/blob/$(commit)/specification/authorization/resource-manager/Microsoft.Authorization/preview/2022-05-01-preview/common-types.json
+
+output-folder: Generated
+
+namespace: Microsoft.Authorization
```
Comment (VeryEarly) [inline src/Resources/Authorization.Management.Sdk/README.md:33]: Microsoft.Azure.Management.Authorization

**Review by VeryEarly — CHANGES_REQUESTED** [review]
the generated SDK doesn't look complete

File: src/Resources/Authorization.Management.Sdk/README.md (hunk 1 above)
Comment (VeryEarly) [reply src/Resources/Authorization.Management.Sdk/README.md]: it's the name of the original SDK: https://github.com/Azure/azure-powershell/blob/main/src/Resources/Resources/Resources.csproj#L19
any reason this should be Microsoft.Authorization?

**Comment by VeryEarly** [comment]
looks like your branch is somehow messed with main branch, I suggest reset your branch or create new branch for this development.

**Comment by mumoryan** [comment]
Closing old prs. Merged one in https://github.com/Azure/azure-powershell/pull/23797

**Pull request closed without merging**